toolkit/tools/scheduler/scheduler.go

```diff
@@ -277,6 +277,26 @@ func startWorkerPool(agent buildagents.BuildAgent, workers, buildAttempts, check
 	return
 }
 
+// debugStuckNode is a debugging function that will print out the stuck node and all nodes that are blocking it.
+func debugStuckNode(buildState *schedulerutils.GraphBuildState, pkgGraph *pkggraph.PkgGraph, stuckNode *pkggraph.PkgNode, indent int) {
+	isStuck := !buildState.IsNodeAvailable(stuckNode)
+	indentSpaces := ""
+	for i := 0; i < indent; i++ {
+		indentSpaces += "  "
+	}
+	if isStuck {
+		logger.Log.Debugf("DEBUG: %s**STUCK** (%s)", indentSpaces, stuckNode.FriendlyName())
+
+		// Iterate over all the nodes that are blocking the stuck node.
+		dependency := pkgGraph.From(stuckNode.ID())
+		for dependency.Next() {
+			dependent := dependency.Node().(*pkggraph.PkgNode)
+			debugStuckNode(buildState, pkgGraph, dependent, indent+1)
+		}
+
+	}
```

Comment:
Since these chains may be long/deep, I'd focus only on the stuck nodes. I think we don't care much about the ones, which were fine and we can make reading this output simpler on the developer.

I've also included one more suggestion about using [string padding](https://golang.cafe/blog/golang-string-padding-example.html).
```suggestion
	const indentWidth = 4

	if buildState.IsNodeAvailable(stuckNode) {
		return
	}

	nodeName := fmt.Sprintf("(%s)", stuckNode.FriendlyName())
	logger.Log.Debugf("%*s", indent * indentWidth, nodeName)

	// Iterate over all the nodes that are blocking the stuck node.
	dependency := pkgGraph.From(stuckNode.ID())
	for dependency.Next() {
		dependent := dependency.Node().(*pkggraph.PkgNode)
		debugStuckNode(buildState, pkgGraph, dependent, indent+1)
	}
```